packages/api/src/routes/medical/__tests__/mapi/e2e/mapi.test.e2e.ts

```diff
@@ -0,0 +1,174 @@
+
+      expect(org).toBeTruthy();
+
+      if (org) {
+        const fhirOrg = await fhirApi.readResource(ResourceType.Organization, org.id, fhirHeaders);
+
+        const cwOrg = await retryFunction<CWOrganization | undefined>(
+          async () => await cwCommands.organization.getOne(org.oid),
+          maxRetries,
+          3000
+        );
+
+        validateLocalOrg(org);
+        validateFhirOrg(fhirOrg, org);
+        validateCWOrg(cwOrg, org);
+      }
+    });
+
+    it("updates an organization", async () => {
+      const org = await medicalApi.getOrganization();
+
+      expect(org).toBeTruthy();
+
+      if (org) {
+        const newName = faker.word.noun();
+
+        const updateOrg: Organization = {
+          ...org,
+          name: newName,
+        };
+
+        const updatedOrg = await medicalApi.updateOrganization(updateOrg);
+
+        await fhirApi.invalidateAll();
+        const fhirOrg: FhirOrg = await fhirApi.readResource(
+          ResourceType.Organization,
+          updatedOrg.id,
+          fhirHeaders
+        );
+
+        const cwOrg: CWOrganization | undefined = await retryFunction<CWOrganization | undefined>(
+          async () => await cwCommands.organization.getOne(updatedOrg.oid),
+          maxRetries,
+          3000,
+          result => result?.name === newName
+        );
+
+        validateLocalOrg(updatedOrg, updateOrg);
+        validateFhirOrg(fhirOrg, updateOrg);
+        validateCWOrg(cwOrg, updateOrg);
+      }
+    });
+
+    it("create a facility", async () => {
+      facility = await medicalApi.createFacility(createFacility);
+
+      validateFacility(facility);
+    });
+
+    it("updates a facility", async () => {
+      const newName = faker.word.noun();
+
+      const updateFacility: Facility = {
+        ...facility,
+        name: newName,
+      };
+
+      facility = await medicalApi.updateFacility(updateFacility);
+
+      validateFacility(facility, updateFacility);
+      expect(facility.name).toBe(newName);
+    });
+
+    it("gets one facility", async () => {
+      const getFacility = await medicalApi.getFacility(facility.id);
+
+      validateFacility(getFacility);
+    });
+
+    it("creates a patient", async () => {
+      patient = await medicalApi.createPatient(createPatient, facility.id);
+
+      const fhirPatient = await fhirApi.readResource(ResourceType.Patient, patient.id, fhirHeaders);
+
+      validateLocalPatient(patient);
+      validateFhirPatient(fhirPatient);
+
+      // Creating a CW patient is done in the background need to await so we can query docs
+      await Util.sleep(10000);
+    });
+
+    it("creates consolidated data for patient", async () => {
+      const payload = createConsolidated(patient.id);
+      const consolidated = await medicalApi.createPatientConsolidated(patient.id, payload);
+      const count = await medicalApi.countPatientConsolidated(patient.id);
+
+      expect(count.total).toBe(1);
```

Comment:
It's good to reference the source for these, in this case check if the `count.total` is equal to `payload.entry?.length`.

Same for other checks on length when possible.